src/ZoneServer/Skills/Handlers/Swordsman/Rodelero/ShieldCharge.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using Melia.Shared.Game.Const;
+using Melia.Shared.L10N;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.Handlers.Common;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Actors.Monsters;
+using Melia.Zone.World.Actors.Pads;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Skills.Handlers.Swordsman.Rodelero
+{
+	/// <summary>
+	/// Handler for the Rodelero skill Shield Charge
+	/// </summary>
+	[SkillHandler(SkillId.Rodelero_ShieldCharge)]
+	public class Rodelero_ShieldCharge : IDynamicCasted
+	{
+		private readonly static TimeSpan DebuffDuration = TimeSpan.FromSeconds(10);
+
+		/// <summary>
+		/// Called when the user starts casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public void StartDynamicCast(Skill skill, ICombatEntity caster)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.SetAttackState(true);
+
+			caster.StartBuff(BuffId.ShieldCharge_Buff, TimeSpan.Zero);
+
+			var pad = new Pad(PadName.Rodelero_ShieldCharge, caster, skill, new Square(caster.Position, caster.Direction, 20, 20));
+			pad.Position = caster.Position;
+			pad.Trigger.LifeTime = TimeSpan.FromSeconds(5);
+			pad.Trigger.MaxActorCount = 3;
+			// TODO: this is clearly impractically low.  We really need an option to just
```

Comment:
This pad needs to be glued to the player's position.  Currently, the update just updates the position to be the player's position every time it fires (which it does very often), but it still feels really laggy and delayed.  It would be much better if we could somehow use a trigger component from the player's position or something.